kyo-core/shared/src/main/scala/kyo/Async.scala

```diff
@@ -170,19 +172,34 @@ object Async:
         def panic[E, A](ex: Throwable): Fiber[E, A] = result(Result.panic(ex))
 
         /** Creates a Fiber from a Future.
+          *
+          * This method allows integration of existing Future-based code with Kyo's Fiber system. It handles successful completion, expected
+          * failures (of type E), and unexpected failures.
           *
           * @param f
-          *   The Future to create a Fiber from
+          *   The Future to convert into a Fiber
+          * @tparam E
+          *   The expected error type that the Future might fail with. Use Throwable if you don't need to catch specific exceptions.
+          * @tparam A
+          *   The type of the successful result
           * @return
           *   A Fiber that completes with the result of the Future
           */
-        def fromFuture[A: Flat](f: Future[A])(using Frame): Fiber[Nothing, A] < IO =
+        def fromFuture[E <: Throwable, A: Flat](f: Future[A])(using
```

Comment:
The API isn't great since users will typically need to specify both type parameters. It might be ok since it seems a low-level API. Alternatively, we could have a `FromFutureOps` class to isolate the `E` type or have a separate `Fiber.fromFutureCatching` method. Suggestions are welcome

Reply:
I think it's reasonable that this returns `Fiber[Throwable, A] < IO`, not `Fiber[E, A]`. Refinement of Aborts is another, but related issue.

Reply:
My initial intuition was that we should avoid replicating error handling logic that should be in `Abort` and not introduce a `fiber.mapFail` method but I think @steinybot made a good point since we already offer `fiber.map/flatMap`. Providing the method can also avoid some of the performance overhead since converting the `Fiber` to a computation requires additional allocations. I think we could always produce a `Fiber[Throwable, A]` here and provide a `fiber.mapFail` method.  @hearnadam wdyt?

Reply:
Thanks for the suggestions folks! I've update the PR to make `Fiber.fromFuture` return `Fiber[Throwable, A]`, introduced `fiber.mapResult` as a more generic method that allows mapping the error, and added a convenience `Async.fromFuture` shortcut.